Homework5: Anton Stychnevsky

Stychnevsky opened the pull request:
Homework # 5

Stychnevsky committed:
Homework # 5

Stychnevsky committed:
fixed codestyle

Stychnevsky committed:
fixed codestyle

Stychnevsky committed:
fixed codestyle

Stychnevsky commented on imdb.py:
```diff
@@ -0,0 +1,99 @@
+    row = next(ratings_file)
+    rates.append(row.split()[2])
+    titles.append(' '.join(row.split()[3:]))
+    years.append(row.split()[-1].strip('()'))
```
Is it pythonic to use 4-5 methods in a row (in one line)? Or better separate it to two lines?

belovmd replied:
we usually use [line.strip for line in lines.split] it's ok. 
but this append looks weird for me and it seems it should be refactored. 
also please avoid magic numbers (2 3)

belovmd commented on imdb.py:
```diff
@@ -0,0 +1,99 @@
+    years_file.write('_____|_________|' + counter * '_' + '\n')
+
+
+def create_histogram(lst, filename='histogram.txt', vertical=True,
```
this functions looks to big it would be better to split it into two (create vertical and usual) and refactor a bit

belovmd commented on imdb.py:
```diff
@@ -0,0 +1,99 @@
+        for key in sorted(dct.keys()):
+            matrix.append(' ' * (max_value - dct[key]) + '#' * dct[key])
+            keys_list.append(str(key) + ' ' * (max_key_len - len(str(key))))
+            ziped = zip(*matrix)
```
i don't understand the purpose of this zip
it seems to me we can avoid it

belovmd commented on runner.py:
```diff
@@ -0,0 +1,19 @@
+    for func in functions:
+        if not func.startswith('__'):
+            if hasattr(pytasks, func):
+                print('{}: {}'.format(func, getattr(pytasks, func)()))
```
it seems we should check if this func callable

belovmd commented on runner.py:
```diff
@@ -0,0 +1,19 @@
+    """
+
+    if not functions:
+        functions = dir(pytasks)
```
we can filter (starts with) it here

belovmd reviewed (commented):
please address these minor issues

Stychnevsky committed:
refactored and improved imbd.py

- delete "magic numbers"
- use str.center() instead of previous "bicycles"
- split "create_histogram" function to vertical and horizontal functions and refactor them
- add "counter" option in vertical histogram function
- use simper method instead of matrix transpose using zip

Stychnevsky committed:
improved runner.py

- add checking if this func callable
- optimize filter startswith()

Stychnevsky committed:
Changed from reversed() to [::-1]

belovmd commented on imdb.py:
```diff
@@ -0,0 +1,96 @@
+for film_title in titles:
+    titles_file.write(film_title + '\n')
+titles_file.close()
+years = map(lambda year: year[:4], years)
```
[year[:4] for year in years}]

belovmd reviewed (commented):
9
